CDP4CrossViewEditor/Generator/CrossviewSheetGenerator.cs

```diff
@@ -0,0 +1,311 @@
+            var displayAlerts = application.DisplayAlerts;
+            var screenupdating = application.ScreenUpdating;
+            var calculation = application.Calculation;
+
+            application.EnableEvents = false;
+            application.DisplayAlerts = false;
+            application.Calculation = XlCalculation.xlCalculationManual;
+            application.ScreenUpdating = false;
+
+            try
+            {
+                application.Cursor = XlMousePointer.xlWait;
+
+                this.crossviewSheet = CrossviewSheetUtilities.RetrieveSheet(workbook, true);
+
+                CrossviewSheetUtilities.ApplyLocking(this.crossviewSheet, false);
+
+                this.PopulateSheetArrays(elementDefinitions, parameterTypes);
+                this.WriteSheet();
+                this.ApplySheetSettings();
+
+                CrossviewSheetUtilities.ApplyLocking(this.crossviewSheet, true);
+
+                this.excelApplication.StatusBar = $"CDP4: Crossview Sheet rebuilt in {sw.ElapsedMilliseconds} [ms]";
+            }
+            catch (Exception ex)
+            {
+                logger.Error(ex);
+                this.excelApplication.StatusBar = $"CDP4: The following error occured while rebuilding the sheet: {ex.Message}";
+            }
+            finally
+            {
+                application.EnableEvents = enabledEvents;
+                application.DisplayAlerts = displayAlerts;
+                application.Calculation = calculation;
+                application.ScreenUpdating = screenupdating;
+
+                application.Cursor = XlMousePointer.xlDefault;
+            }
+        }
+
+        /// <summary>
+        /// Apply formatting settings to the Parameter sheet
+        /// </summary>
+        private void ApplySheetSettings()
+        {
+            var outline = this.crossviewSheet.Outline;
+            outline.AutomaticStyles = false;
+            outline.SummaryRow = XlSummaryRow.xlSummaryAbove;
+            outline.SummaryColumn = XlSummaryColumn.xlSummaryOnRight;
+
+            this.excelApplication.ActiveWindow.DisplayGridlines = false;
+        }
+
+        /// <summary>
+        /// collect the information that is to be written to the Parameter sheet
+        /// </summary>
+        /// <param name="elementDefinitions"></param>
+        /// <param name="parameterTypes"></param>
+        private void PopulateSheetArrays(IEnumerable<ElementDefinition> elementDefinitions, IEnumerable<ParameterType> parameterTypes)
+        {
+            // Instantiate the different rows
+            var assembler = new CrossviewSheetRowAssembler();
+            assembler.Assemble(elementDefinitions);
+            var excelRows = assembler.ExcelRows;
+
+            // Use the instantiated rows to populate the excel array
+            var parameterArrayAssembler = new CrossviewArrayAssembler(excelRows, parameterTypes);
+            this.parameterContent = parameterArrayAssembler.ContentArray;
```

Comment:
This should be called "bodyContent" - it's no longer related to parameters, and it seems fitting to be a generic property name that would fit anywhere (hence why I am not suggesting "crossviewContent").